argopy/utils/transform.py

```diff
@@ -352,24 +354,48 @@ def split_data_mode(ds: xr.Dataset) -> xr.Dataset:
     -------
     :class:`xr.Dataset`
     """
+    if ds.argo._type != "profile":
```

Comment:
Raise an error to avoid passing through for the wrong reasons and not detecting it